addons/binding/org.openhab.binding.blebox/src/main/java/org/openhab/binding/blebox/handler/DimmerHandler.java

```diff
@@ -0,0 +1,114 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.blebox.handler;
+
+import static org.openhab.binding.blebox.BleboxBindingConstants.CHANNEL_BRIGHTNESS;
+
+import java.math.BigDecimal;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.blebox.devices.Dimmer;
+import org.openhab.binding.blebox.internal.BleboxDeviceConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DimmerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Szymon Tokarski - Initial contribution
+ */
+public class DimmerHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(DimmerHandler.class);
+    private Dimmer dimmer;
+
+    Runnable runnable = new Runnable() {
+        @Override
+        public void run() {
+            try {
+                if (dimmer != null) {
+                    PercentType brightness = dimmer.getBrightness();
+
+                    if (brightness != null) {
+                        updateState(CHANNEL_BRIGHTNESS, brightness);
+
+                        if (getThing().getStatus() == ThingStatus.OFFLINE) {
+                            updateStatus(ThingStatus.ONLINE);
+                        }
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE);
+                    }
+                }
+            } catch (Exception e) {
+                logger.info("Polling device state failed: {}", e.toString());
+            }
+        }
+    };
+    private ScheduledFuture<?> pollingJob;
+
+    public DimmerHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_BRIGHTNESS)) {
+            if (command instanceof PercentType) {
+                dimmer.setBrightness((PercentType) command);
+            } else if (command instanceof OnOffType) {
+                if (((OnOffType) command).equals(OnOffType.ON)) {
+                    dimmer.setBrightness(PercentType.HUNDRED);
+                } else {
+                    dimmer.setBrightness(PercentType.ZERO);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final String ipAddress = (String) getConfig().get(BleboxDeviceConfiguration.IP);
+
+        if (ipAddress != null) {
```

Comment:
Because `ipAddress` is required this value will never be null as it's check by openHAB before. This method will never be called when it's null. So this check is not really needed in my opinion.